### PR TITLE
VCST-1808: back in stock module

### PR DESCRIPTION
**Pull request by mihey8800**
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-2.7.0-pr-1358-d8ae-d8ae2f26.zip

### CODE REVIEW

File: client-app/core/utilities/line-items/index.ts [hunk 1]
```diff
@@ -36,18 +36,22 @@ export function groupByVendor<T extends LineItemType | OrderLineItemType>(items:
   return result;
 }
 
-function prepareItemPrices(item: AnyLineItemType) {
-  const price = "price" in item ? item.price : undefined;
+function prepareItemPrices(item: AnyLineItemType | Product) {
```
Comment (NaMax66) [inline client-app/core/utilities/line-items/index.ts:39]: extend AnyLineItemType.

File: client-app/modules/quotes/useUserQuotes.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import type { QuoteType } from "../quotes/api/graphql/types";
 import type { ISortInfo } from "@/core/types";
 import type { Ref } from "vue";
 
-const DEFAULT_ITEMS_PER_PAGE = 10;
+const DEFAULT_ITEMS_PER_PAGE = 9999;
```
Comment (NaMax66) [inline client-app/modules/quotes/useUserQuotes.ts:9]: return 10

File: client-app/shared/back-in-stock/components/back-in-stock-notify-button.vue [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      @click="updateBackInStockSubscription"
+    >
+      {{ isSubscribed ? $t("common.messages.you_will_be_notified") : $t("common.messages.notify_me_when_in_stock") }}
+    </VcButton>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue";
+import { globals } from "../../../core/globals";
```
Comment (NaMax66) [inline client-app/shared/back-in-stock/components/back-in-stock-notify-button.vue:19]: import starts from \@

File: client-app/shared/catalog/components/product/variations-table.vue [hunk 4]
```diff
@@ -229,24 +229,24 @@ const mappedLineItems = computed(() => {
   return mapped;
 });
 
-function getStockQuantity(variation: VariationType) {
+function getStockQuantity(variation: Product) {
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product/variations-table.vue:232]: should be variation type

File: client-app/pages/account/back-in-stock-subscriptions.vue [hunk 5]
```diff
@@ -0,0 +1,201 @@
+<template>
+  <div>
+    <VcLoaderOverlay :visible="loading" fixed-spinner />
+
+    <BackButtonInHeader v-if="isMobile" @click="$router.back()" />
```
Comment (NaMax66) [inline client-app/pages/account/back-in-stock-subscriptions.vue:5]: should work with $router.back without ()

File: client-app/pages/account/back-in-stock-subscriptions.vue [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+        <VcTypography tag="h1">
+          {{ $t("shared.account.navigation.links.back_in_stock_subscriptions") }}
+        </VcTypography>
+      </div>
+
+      <div ref="listElement" class="mt-5 w-full">
+        <!-- Skeletons -->
+        <template v-if="!loading">
+          <div v-if="isMobile" class="grid grid-cols-2 gap-x-4 gap-y-6">
+            <ProductSkeletonGrid v-for="i in actualPageRowsCount" :key="i" />
```
Comment (NaMax66) [inline client-app/pages/account/back-in-stock-subscriptions.vue:19]: rename i -> index

File: client-app/shared/back-in-stock/components/back-in-stock-subscriptions-line-items.vue [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        :list-price="item.listPrice"
+        :actual-price="item.actualPrice"
+        :total="item.extendedPrice"
+        :disabled="pendingItems[item.id]"
+        :deleted="item.deleted"
+        with-image
+        with-properties
+        with-price
+        removable
+        @remove="() => removeSingleItem(item.id)"
```
Comment (NaMax66) [inline client-app/shared/back-in-stock/components/back-in-stock-subscriptions-line-items.vue:33]: () => removeSingleItem(item.id) -> removeSingleItem(item.id)

File: client-app/shared/back-in-stock/composables/useBackInStockSubscriptions.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+import type {
+  ActivateBackInStockSubscriptionCommandType,
+  BackInStockSubscriptionQueryType,
+  BackInStockSubscriptionType,
+  DeactivateBackInStockSubscriptionCommandType,
+} from "@/core/api/graphql/types";
+import type { ISortInfo } from "@/core/types";
+import type { Ref } from "vue";
+
+const DEFAULT_ITEMS_PER_PAGE = 9999;
```
Comment (NaMax66) [inline client-app/shared/back-in-stock/composables/useBackInStockSubscriptions.ts:19]: why so large? Why not pagination?

File: client-app/pages/account/back-in-stock-subscriptions.vue [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+      props: {
+        productId: item.id,
+        productName: item.name,
+
+        async onResult(): Promise<void> {
+          const previousPagesCount = pagesCount.value;
+          if (previousPagesCount > 1 && previousPagesCount === page.value && previousPagesCount > pagesCount.value) {
+            page.value -= 1;
+          }
+          await fetchProductsAndSubscriptions();
```
Comment (NaMax66) [inline client-app/pages/account/back-in-stock-subscriptions.vue:177]: void fetchProductsAndSubscriptions()
no need to wait

File: client-app/pages/account/back-in-stock-subscriptions.vue [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+const pagedListItems = computed<PreparedLineItemType[]>(() =>
+  preparedLineItems.value.slice((page.value - 1) * itemsPerPage.value, page.value * itemsPerPage.value),
+);
+
+const actualPageRowsCount = computed<number>(() => pagedListItems.value.length || itemsPerPage.value);
+
+const isMobile = breakpoints.smaller("lg");
+
+const fetchProductsAndSubscriptions = async () => {
+  itemsPerPage.value = 9999;
```
Comment (NaMax66) [inline client-app/pages/account/back-in-stock-subscriptions.vue:131]: use constant for pagination. and not so large

File: client-app/pages/account/back-in-stock-subscriptions.vue [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+  await fetchSubscriptions();
+  if (backInStockSubscriptions.value.length > 0) {
+    await fetchProducts({ productIds: backInStockSubscriptions.value.map((item) => item.id!), itemsPerPage: 9999 });
```
Comment (NaMax66) [inline client-app/pages/account/back-in-stock-subscriptions.vue:134]: use constant for pagination. and not so large

**Review by NaMax66 — CHANGES_REQUESTED** [review]

File: client-app/shared/back-in-stock/composables/useBackInStockSubscriptions.ts (hunk 8 above)
Comment (mihey8800) [reply client-app/shared/back-in-stock/composables/useBackInStockSubscriptions.ts:19]: buisness rule

File: client-app/pages/account/back-in-stock-subscriptions.vue [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+          <div v-else class="flex flex-col rounded border bg-additional-50 shadow-sm">
+            <WishlistProductItemSkeleton v-for="i in actualPageRowsCount" :key="i" class="even:bg-neutral-50" />
+          </div>
+        </template>
+
+        <!-- List details -->
+        <template v-if="!!subscriptionsItems.length && !loading">
+          <VcWidget size="lg">
+            <div class="flex flex-col gap-6">
+              <WishlistLineItems
```
Comment (goldenmaya) [inline client-app/pages/account/back-in-stock-subscriptions.vue:31]: back-in-stock-subscriptions-line-items?

**Comment by NaMax66** [comment]
outdated

**Pull request closed without merging**